src/user/user.controller.ts

```diff
@@ -45,16 +45,15 @@ export class UserController {
   @Get('/me')
   @UseGuards(FirebaseAuthGuard)
   async getUserByFirebaseId(@Req() req: Request): Promise<GetUserDto> {
-    const user = req['user'];
-    this.userService.updateUser({ lastActiveAt: new Date() }, user.user.id);
-    return user as GetUserDto;
+    await this.userService.updateUser({ lastActiveAt: new Date() }, req['userEntity'].id);
+    return (await this.userService.getUserProfile(req['userEntity'].id)).userDto;
```

Comment:
the lack of await here was intentional - we async update the users `lastActiveAt` after they have called their user profile.

i don't think we need the new `getUserProfile` function here, the change required is only
```
  async getUserByFirebaseId(@Req() req: Request): Promise<UserEntity> {
    const user = req['userEntity'];
    this.userService.updateUser({ lastActiveAt: new Date() }, user.id);
    return formatUserObject(user);
```

Reply:
Sorry to double back on this @SangilYun. @annarhughes I think we do need the getUserProfile here.  As far as I understand, this endpoint is how we get all the user details to populate frontend state. Do you have a different understanding?

Reply:
Anna is away so i'll update you on this @SangilYun - sorry for the delay. I would to keep the service method `getUserProfile`. The reason this change still works is because the firebaseAuthguard is still pulling too much data. If you look at getUserByFirebaseId which the firebaseAuthguard calls, it;s getting subscriptions, sessions, partnerAdmin data. We just don't need that on every request. So I'd like the getUserProfile to be the function that adds all this contextual data. Does that make sense? Then after this change, I can return to the authGuard and only pull the data from the user repository. This means we can drop all the complex module dependencies. I hope that makes sense and sorry for the confusion in the team!

Reply:
@eleanorreem Appreciate the feedback! I've added the service function back in, please have a look :)